packages/chaire-lib-backend/src/utils/processManagers/__tests__/TrRoutingProcessManager.test.ts

```diff
@@ -78,11 +81,72 @@ describe('TrRouting Process Manager: start', () => {
             'trRouting4000',
             'trRouting',
             './trRouting',
-            ['--port=4000', `--osrmPort=${walkingOsrmMode.getHostPort().port}`, `--osrmHost=${walkingOsrmMode.getHostPort().host}`, '--debug=0', `--cachePath=${directoryManager.projectDirectory}/cache/test`],
+            ['--port=4000', `--osrmPort=${walkingOsrmMode.getHostPort().port}`, `--osrmHost=${walkingOsrmMode.getHostPort().host}`, '--debug=0', `--cachePath=${directoryManager.projectDirectory}/cache/test`, '--threads=2'],
             'ready.',
             false,
             __dirname,
             false
         );
     });
-});
\ No newline at end of file
+    test('start batch process with 1 cpu', async () => {
+        const status = await TrRoutingProcessManager.startBatch(1);
+        expect(status).toEqual({
+            status: 'started',
+            service: 'trRoutingBatch',
+            port: 14000
+        });
+        expect(startProcessMock).toHaveBeenCalledTimes(1);
+        expect(startProcessMock).toHaveBeenCalledWith(
+            'trRouting14000',
+            'trRouting',
+            'trRouting',
+            ['--port=14000', `--osrmPort=${walkingOsrmMode.getHostPort().port}`, `--osrmHost=${walkingOsrmMode.getHostPort().host}`, '--debug=0', `--cachePath=${directoryManager.projectDirectory}/cache/test`],
+            'ready.',
+            false,
+            undefined,
+            false
+        );
+    });
+    test('start batch process with 4 cpus', async () => {
+        const status = await TrRoutingProcessManager.startBatch(4);
+        expect(status).toEqual({
+            status: 'started',
+            service: 'trRoutingBatch',
+            port: 14000
+        });
+        expect(startProcessMock).toHaveBeenCalledTimes(1);
+        expect(startProcessMock).toHaveBeenCalledWith(
+            'trRouting14000',
+            'trRouting',
+            'trRouting',
+            ['--port=14000', `--osrmPort=${walkingOsrmMode.getHostPort().port}`, `--osrmHost=${walkingOsrmMode.getHostPort().host}`, '--debug=0', `--cachePath=${directoryManager.projectDirectory}/cache/test`, '--threads=4'],
+            'ready.',
+            false,
+            undefined,
+            false
+        );
+    });
+    test('start batch process with 4 cpus with 2 limit', async () => {
+        // Override the configuration
+        // TODO might have a better way to do this
+        config.maxParallelCalculators = 2;
+
+        const status = await TrRoutingProcessManager.startBatch(4);
```

Comment:
I know you are really testing the number of threads, but might be worth adding a test with a different port and cache directory while you're here.